src/api/composition/CompositionCollection.js

```diff
@@ -189,6 +195,15 @@ define([
 
             this.provider.add(this.domainObject, child.identifier);
         } else {
+            if (this.returnMutables && this.publicAPI.objects.supportsMutation(child)) {
+                let keyString = this.publicAPI.objects.makeKeyString(child.identifier);
+
+                if (this.publicAPI.objects.supportsMutation(child)) {
```

Comment:
Wouldn't this always be true given the outer if block?

Reply:
You appear to be correct. Let me take a look and de-crappify accordingly 😄